torch/lib/c10d/ProcessGroupNCCL.hpp

```diff
@@ -136,6 +136,8 @@ class ProcessGroupNCCL : public ProcessGroup {
 
     // Store a reference to NCCL collective's outputs to be used by getFuture.
     std::shared_ptr<std::vector<at::Tensor>> outputs_;
+    // Store a reference to the NCCL stream that runs all FutureNCCL callbacks.
+    std::shared_ptr<at::cuda::CUDAStream> futureNCCLCallbackStream_;
```

Comment:
Pass a vector here for all the streams for all devices. We can have the limitations for single process multi device mode only in FutureNCCL where its necessary.